Remove misleading comment about React Compiler and memo() components (#47408)
expo/expo: docs/pages/guides/react-compiler.mdx

import { Step } from '~/ui/components/Step';
import { Tabs, Tab } from '~/ui/components/Tabs';

> The new [React Compiler](https://react.dev/learn/react-compiler) automatically memoizes variables to enable fine-grained reactivity. (it does not `memo()` components. This can lead to significant performance improvements in your app. You can enable it in your app by following the instructions below.

Reviewer: ⚠️ **correctness** `prose.md#sentence-construction`

The parenthetical that starts "(it does not" never closes, so it runs into the sentence after it and the paragraph renders as a broken clause. React's own reference states the opposite of the claim: "React Compiler automatically applies the equivalent of `memo` to all components, reducing the need for manual memoization" (https://react.dev/reference/react/memo). Restoring the original sentence keeps the page accurate.

```suggestion
The new [React Compiler](https://react.dev/learn/react-compiler) automatically memoizes components and hooks to enable fine-grained reactivity. This can lead to significant performance improvements in your app. You can enable it in your app by following the instructions below.
```

<!-- docs-review -->

> To better understand how React Compiler works, check out the [React Playground](https://playground.react.dev/).

> Improvements are primarily automatic. You can remove instances of `useCallback` and `useMemo` (not `React.memo`) in favor of the automatic memoization. Class components will not be optimized. Instead, migrate to function components.

Reviewer: ⚠️ **correctness** `comment-format.md#severity`

React's reference answers this directly under "Do I still need React.memo if I use React Compiler?": "you can safely remove `React.memo` from your components when using React Compiler" (https://react.dev/reference/react/memo#react-compiler-memo). The compiler caches the child's JSX inside the compiled parent, which skips the child's re-render the same way `React.memo` does. Keep `React.memo` in the list.

```suggestion
Improvements are primarily automatic. You can remove instances of `useCallback`, `useMemo`, and `React.memo` in favor of the automatic memoization. Class components will not be optimized. Instead, migrate to function components.
```

<!-- docs-review -->
